magma/primitives/memory.py

```diff
@@ -31,9 +32,10 @@ def __init__(self, depth, width, init=None):
                                "has_init": init is not None}
         self.coreir_configargs = {}
         if init is not None:
-            raise NotImplementedError("CoreIR's memory primitive does not "
-                                      "compile init arg to verilog yet")
-            # self.coreir_configargs["init"] = <convert to json arg>
+            init_bv = BitVector[width](init[0])
+            for i, elem in enumerate(init[1:]):
+                init_bv = init_bv.concat(BitVector[width](elem))
+            self.coreir_configargs["init"] = init_bv
```

Comment:
```suggestion
            functools.reduce(BitVector[width].concat, (BitVector[width](i) for i in init))
            init_bv = BitVector[width](init[0])
            for i, elem in enumerate(init[1:]):
                init_bv = init_bv.concat(BitVector[width](elem))
            self.coreir_configargs["init"] = init_bv
```
(import `functools` of course)